### ISSUE TITLE
XGBClassifier' object has no attribute 'n_classes_'

### ISSUE THREAD

**Issue by fxc123** [TeamHG-Memex/eli5#204]
I want to explain a XGB multiclasses model (using softmax) by eli, 

```
import eli5 as eli
eli.show_weights(clf2)
eli.show_prediction(clf2,arr[3],show_feature_values=True)

```

but get the following error:

XGBClassifier' object has no attribute 'n_classes_'

**Comment by fxc123** [comment]
I use python3 & latest xgboost version

**Comment by lopuhin** [comment]
Hello @fxc123 , is ``clf2`` an ``xgboost.Booster`` instance, or an ``xgboost.XGBClassifier``? It it's an  an ``xgboost.Booster``, then this case is not yet supported (see https://github.com/TeamHG-Memex/eli5/issues/193)

**Comment by lopuhin** [comment]
ah sorry, it must be something else... could you please share the full traceback, and if possible, the way you construct the classifier?

**Comment by fxc123** [comment]
@lopuhin clf2 is XGBClassifer.

callstack is shown below:

```
AttributeError                            Traceback (most recent call last)
<ipython-input-94-49c240fafeec> in <module>()
      5 
      6 
----> 7 eli.show_prediction(clf2,arr[3],show_feature_values=True)
      8 
      9 #eli.show_prediction(clf2,arr[3])

/usr/local/lib/python3.6/site-packages/eli5/ipython.py in show_prediction(estimator, doc, **kwargs)
    261     """
    262     format_kwargs, explain_kwargs = _split_kwargs(kwargs)
--> 263     expl = explain_prediction(estimator, doc, **explain_kwargs)
    264     html = format_as_html(expl, **format_kwargs)
    265     return HTML(html)

/usr/local/lib/python3.6/site-packages/singledispatch.py in wrapper(*args, **kw)
    208 
    209     def wrapper(*args, **kw):
--> 210         return dispatch(args[0].__class__)(*args, **kw)
    211 
    212     registry[object] = func

/usr/local/lib/python3.6/site-packages/eli5/xgboost.py in explain_prediction_xgboost(xgb, doc, vec, top, top_targets, target_names, targets, feature_names, feature_re, feature_filter, vectorized)
    139     proba = predict_proba(xgb, X)
    140     scores_weights = _prediction_feature_weights(
--> 141         xgb, X, feature_names, xgb_feature_names)
    142 
    143     x = get_X0(add_intercept(X))

/usr/local/lib/python3.6/site-packages/eli5/xgboost.py in _prediction_feature_weights(xgb, X, feature_names, xgb_feature_names)
    188         _target_feature_weights,
    189         feature_names=feature_names, xgb_feature_names=xgb_feature_names)
--> 190     n_targets = _xgb_n_targets(xgb)
    191     if n_targets > 1:
    192         # For multiclass, XGBoost stores dumps and leaf_ids in a 1d array,

/usr/local/lib/python3.6/site-packages/eli5/xgboost.py in _xgb_n_targets(xgb)
    253 def _xgb_n_targets(xgb):
    254     if isinstance(xgb, XGBClassifier):
--> 255         return 1 if xgb.n_classes_ == 2 else xgb.n_classes_
    256     elif isinstance(xgb, XGBRegressor):
    257         return 1

AttributeError: 'XGBClassifier' object has no attribute 'n_classes_'
```

**Comment by fxc123** [comment]
And I construct the clasiffer like this:

```
bst = xgb.Booster({'nthread':4})
bst.load_model("/Users/lucas/risk/model")
clf2 = XGBClassifier()
clf2._Booster = bst
```

**Comment by jnothman** [comment]
did you fit the model?

**Comment by fxc123** [comment]
I load a model from a existed model file. So I think I don't need to call fit any more?

**Comment by lopuhin** [comment]
Looks like this ``clf2._Booster = bst`` is not enough to convert a model to the sklearn interface, at least if the booster is loaded from disk. There might be some hacky way around it, but if this is feasible, I think it's best to either wait for #193 or train a new model using sklearn API (``XGBClassifier`` or ``XGBRegressor``).

**Comment by lopuhin** [comment]
@fxc123 now that #193 is fixed in master, you can pass booster object to show_prediction and show_weights

**Comment by williamty** [comment]
i met the same problem, my xgboost is 2.0.0:
`booster = xgb.Booster(model_file="F:\\mljar\\AutoML_NM2\\43_Xgboost\\learner_fold_4.xgboost")
xgb_classifier = XGBClassifier()
xgb_classifier._Booster = booster

predict_function=lambda z: xgb_classifier.predict_proba(TabularTransform().transform(z))

explainer = GPTExplainer(
    training_data=tabular_data,
    predict_function=predict_function,
    apikey="sk-xxxxxx"
)`

**Comment by williamty** [comment]
> i met the same problem, my xgboost is 2.0.0: `booster = xgb.Booster(model_file="F:\mljar\AutoML_NM2\43_Xgboost\learner_fold_4.xgboost") xgb_classifier = XGBClassifier() xgb_classifier._Booster = booster
> 
> predict_function=lambda z: xgb_classifier.predict_proba(TabularTransform().transform(z))
> 
> explainer = GPTExplainer( training_data=tabular_data, predict_function=predict_function, apikey="sk-xxxxxx" )`

i have figured it out. use this:
xgb_classifier = XGBClassifier()
xgb_classifier.load_model("F:\\mljar\\AutoML_NM2\\43_Xgboost\\learner_fold_4.xgboost")
